hpOneView/resources/security/certificates_server.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+###
+# (C) Copyright [2020] Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+from future import standard_library
+
+standard_library.install_aliases()
+
+
+from hpOneView.resources.resource import Resource
+
+
+class CertificatesServer(Resource):
+    """
+    The Certificates Servers resource provides REST APIs for configuration of device or
+    server certificates for the appliance to establish SSL communication with other managed network entities.
+
+    Import, Update and Delete APIs are asynchronous and GET API is synchronous.
+
+    """
+    URI = '/rest/certificates'
```

Comment:
the resource base uri is /rest/certificates/server. So, not sure if its ok as per design of other resources in this sdk.

Reply:
in this scenario, the expected uri for getting remote certificate is /rest/certificates/https/remote/{example) which is different from /rest/certificates/server. 
And in ResourceHelper class we are validating all the uris by using below function.

    def validate_resource_uri(self, path):
        """Helper method to validate URI of the resource."""
        if self._base_uri not in path:
            logger.exception('Get by uri : unrecognized uri: (%s)' % path)
            raise exceptions.HPOneViewUnknownType(UNRECOGNIZED_URI)

Which means it checks if base_uri is present in the uri provided or not. If base uri is not present then the call fails with UNRECOGNIZED_URI error.

So if I set base uri as /rest/certificates/server, then get_remote certificate function fails at above validation.

Reply:
Validation of resource uri is correct way to do check the resource uri. That's why i am not sure why OneView has kept this endpoint in server certificates as we have appliance certificates for which /certificates/https is the named context. Anyways keeping this discussion aside, lets keep it as OneView has design it.

Reply:
Thanks. Kept the function as it is.